libs/shared/ui-components/src/SocialCard/SocialCard.tsx

```diff
@@ -18,6 +18,15 @@ export const SocialCard: FC<TSocialCardProps> = ({
 }) => {
   const { asPath } = useRouter();
 
+  /**
+   * url is constructed starting from https:// because 'NEXT_PUBLIC_VERCEL_URL' doesn't provide the protocol(https)
+   * and for og:url requires absolute URL which includes the protocol and domain name
+   * using the prefix NEXT_PUBLIC is used to indicate that the environment variable is public,
+   * meaning that it can be accessed by the browser.
+   * 'NEXT_PUBLIC_VERCEL_URL' is set by Vercel when the application is deployed, and it can be accessed by the application code.
+   * This kind of URL is need because of og:url needs absolute path and allows it to use this as the correct canonical tag
+   * Also twitter doesn't have something similar to og:url like twitter:url.
+   */
```

Comment:
@bskinn What about this comments should I remove the /** or do we need it in the doc. What do you think?

Reply:
I would convert this to `/*`, because it's a comment on an internal variable, not part of the external API of the component. (A user of the component doesn't need to know about how `url` is constructed; this comment is here to help a future developer who is looking into the innards of the component to see how it works.